### PR TITLE
Add micro framework CarouselView

### PR DESCRIPTION
**Pull request by mrylmz**

### CODE REVIEW

File: Modules/CarouselView/CarouselView.swift [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+        if model.pages.count < contentViews.count {
+            let removedIndices = model.pages.count ..< contentViews.count
+            contentViews[removedIndices].forEach { $0.removeFromSuperview() }
+            contentViews.removeSubrange(removedIndices)
+        } else if contentViews.count < model.pages.count {
+            let addedIndices = contentViews.count ..< model.pages.count
+            addedIndices.forEach { _ in
+                let view: ContentView = .instantiate()
+                contentViews.append(view)
```
Comment (SimonNumberTwo) [inline Modules/CarouselView/CarouselView.swift]: What is the purpose of adding empty views? Shouldn't we add more content views from the model? Or when is this condition even met? For me adding empty views currently makes no sense.

File: Modules/CarouselView/CarouselView.swift [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            }
+        }
+
+        contentViews.enumerated().forEach { index, view in
+            view.model = model.pages[index]
+        }
+
+        if !model.pages.isEmpty {
+            leftContentViewPlaceholder.model = model.pages[modIndex(model.pages.count - 1, model.pages.count)]
+            rightContentViewPlaceholder.model = model.pages[0]
```
Comment (SimonNumberTwo) [inline Modules/CarouselView/CarouselView.swift]: Are you sure about index zero here? Shouldn't zero be the one cantered? Doesn't it have to be flexible?

File: Modules/CarouselView/CarouselView.swift [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+    public func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        guard isScrollViewDidScrollObservationEnabled else { return }
+
+        let contentSize = scrollView.contentSize.width - bounds.width * 2
+        let contentOffset = scrollView.contentOffset.x - bounds.width
+        let newContentViewIndex = bounds.width > 0 ? modIndex((Int(round(contentOffset / bounds.width))), model.pages.count) : 0
+        if contentViewIndex != newContentViewIndex {
+            contentViewIndex = newContentViewIndex
+            print(contentViewIndex)
```
Comment (SimonNumberTwo) [inline Modules/CarouselView/CarouselView.swift]: Please remove the `print` here.

File: Modules/CarouselView/CarouselView.swift (hunk 1 above)
Comment (mrylmz) [reply Modules/CarouselView/CarouselView.swift]: This is balancing the number of contentViews with the number of pages provided in the model. The models then become assigned afterwards.

File: Modules/CarouselView/CarouselView.swift (hunk 2 above)
Comment (mrylmz) [reply Modules/CarouselView/CarouselView.swift]: Yes that is correct here we do add the first page to the end of the list

**Pull request merged**